File: Circular.java
import java.util.Scanner;
class Node{
    Node next;
    int data;
    public Node(int data){
        this.data=data;
        next=null;
    }
}
class Link{
    Node insert(int data,Node head){
        Node temp=new Node(data);
        if(head==null){
            head=temp;
            return head;
        }
        else{
            Node t=head;
            while(t.next!=null){
                t=t.next;
            }
            t=temp;
        return head;
        }
    }
    void display(Node head){
        if(head==null){
            System.out.println("empty linkedlist");
        }
        else{
            Node temp=head;
            while(temp.next!=null){
                System.out.println(temp.data);
                temp=temp.next;
            }
        }
    }
}
public class Circular{
    public static void main(String[] args){
        Link ob=new Link();
        Node head=null;
        head=ob.insert(13,head);
        ob.display(head);
    }
}

File: circular.c
#include<stdio.h>
#include<stdlib.h>
struct Node{
	int data;
	struct Node* next;
};
struct Node *create(){
	struct Node *temp=(struct Node*)malloc(sizeof(struct Node));
	return temp;
}
struct Node *insert(struct Node *last){
	struct Node *temp=create();
	printf("enter data\n");
	scanf("%d",&temp->data);
	if(last==NULL){
		last=temp;
		last->next=last;
	}
	else{
		temp->next=last->next;
		last->next=temp;
		last=temp;
	}
	return last;
}
void display(struct Node *last){
	if(last==NULL){
		printf("empty");
	}
	else{
		struct Node *temp=last->next;
		do{
			printf("%d\n",temp->data);
			temp=temp->next;
		}while(temp!=last->next);
	}
}
int main(){
	struct Node *last=NULL;
	last=insert(last);
	last=insert(last);
	last=insert(last);
	display(last);
	return 0;
}
